### DISCUSSION TITLE
Retrieving underlying table/view/entity from field

### DISCUSSION THREAD

**Discussion by ghost** [jOOQ/jOOQ#15914] [Q&A] [answered]
Hello,

Consider the same query as in the discussion post I made earlier today:

```
            final var accounts = table("accounts").as("a");
            final var transactions = table("transactions").as("t");
            final var query = context
                    .select(field("a.id"), field("t.*"))
                    .from(accounts)
                    .join(transactions)
                    .on(field("a.id").eq(field("t.account_id")))
                    .fetch();
```

I'd like to find out what table/view the field "a.id" belongs to. Is there a way to do so? I see that there is a TableField/TableFieldImpl that does have a way of retrieving the table. But that is only meant for generated code. I tried retrieving the associated metadata via context.meta() but apparently there is no way of retrieving columns directly for a table? Any ideas would be appreciated.

**Comment by lukaseder** [comment, marked as answer]
You could use identifiers instead of plain SQL templates:
https://www.jooq.org/doc/latest/manual/sql-building/names/

Of course, using the code generator is even better for hundreds of reasons:
https://blog.jooq.org/why-you-should-use-jooq-with-code-generation/

But you probably have a dynamic schema preventing you from using it?

**Reply by ghost** [reply]
It's more of a buy-in issue. I don't have any control on what features of jOOQ the end users are going to use. Some might not use code generation at all.

Thank you for the info in regards to the identifiers, I'll just tell them to use that for now.

**Reply by lukaseder** [reply]
> It's more of a buy-in issue. I don't have any control on what features of jOOQ the end users are going to use. Some might not use code generation at all.

Thanks for clarifying. You probably told me already, I just have to point it out on every issue, because there are many people who aren't aware of code generation, possibly finding this discussion from google, in the future.
